exercises/practice/robot-name/.meta/src/reference/java/Robot.java

```diff
@@ -1,18 +1,26 @@
+import java.util.HashSet;
 import java.util.Random;
+import java.util.Set;
 
 public class Robot {
 
     private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
     private String name;
     private final Random random;
+    private static Set<String> nameLog = new HashSet<>();
 
     public Robot() {
         random = new Random();
         assignNewName();
     }
 
     private void assignNewName() {
-        name = String.format("%s%d", prefix(), suffix());
+        String newName = String.format("%s%d", prefix(), suffix());
+        if (nameLog.add(newName)) {
+            this.name = newName;
+        } else {
+            assignNewName();
```

Comment:
This structure actually has really bad recursion after you have generated most of the names and will cause a stack overflow eventually. I recommend making this a for loop instead.

Reply:
Thanks for the suggestion! I'll change the code to a loop.